Sampler/scripts/Set-SamplerTaskVariable.ps1

```diff
@@ -95,7 +95,7 @@ if (-not [string]::IsNullOrEmpty($BuiltModuleSubDirectory))
 {
     $BuiltModuleSubdirectory = Get-SamplerAbsolutePath -Path $BuiltModuleSubDirectory -RelativeTo $OutputDirectory
 }
-elseif ($BuildInfo.ContainsKey('BuiltModuleSubdirectory'))
+elseif ($BuildInfo -and $BuildInfo.ContainsKey('BuiltModuleSubdirectory'))
```

Comment:
I just added a check so that $BuildInfo is not null before using method `ContainsKey()`, it would otherwise throw an error if using `Invoke-Pester` when using task `test` thorugh Invoke-Build this is always set to a value to method `ContainsKey` will always work. I don't see how it will change any logic, but I could add more Code to Pester to always pass `BuildInfo = @{}` so I can revert this change.

Reply:
Reverted the change.

Reply:
There was already code in the tests to handle it, so I could just revert the change.

Reply:
Thanks, the logic change is that it's best for the variables to always have an absolute path.
Basically, you always want to use $builtSubmoduleDirectory for things that targets your Built module base, so that even if it's not set, it resolves to something, which is the same as the $OutputDirectory if not set...